### ISSUE TITLE
Focus the headline after a modal opens

### ISSUE THREAD

**Issue by darekkay** [darekkay/dashboard#6]
After opening a modal (e.g. Settings or Widget Configuration), the headline should be focused (if one exists) for better accessibility. Additionally, the DOM order of the headline and close button should be reversed (while visually maintained using CSS `order`), so the modal close button is focused last.

- See also: [WAI-ARIA Authoring Practices](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal)

**Comment by darekkay** [comment]
There is [no fixed rule](https://adrianroselli.com/2020/10/dialog-focus-in-screen-readers.html) where the focus should be moved to after opening a modal. There is an [ongoing discussion](https://github.com/whatwg/html/pull/4184) about the default behavior for the HTML `dialog` element. Currently, it is the modal itself, which is a sufficient solution. 

The close button being focused first does follow the default browser focus order. It also makes sense for users wanting to close the modal without having to go through the whole content (even though it's possible with Escape key as well).
